code/modules/reagents/chemistry/reagents/alcohol_reagents.dm

```diff
@@ -2356,6 +2356,219 @@ All effects don't start immediately, but rather get worse over time; the rate is
+	if(iszombie(M) || isvampire(M) || isdullahan(M)) //Rare races!
+		quality = RACE_DRINK
+	else
+		M.adjust_disgust(25)
+	return ..()
+
+/datum/reagent/consumable/ethanol/abduction_fruit
+	name = "Abduction Fruit"
+	color = "#DEFACD" //(222, 250, 205)
+	description = "Mixing of juices to make an alien taste."
+	boozepwr = 80 //Strong
+	taste_description = "grass and lime"
+	glass_icon_state = "abduction_fruit"
+	glass_name = "glass of Abduction Fruit"
+	glass_desc = "Mixed fruits that were never ment to be mixed..."
+
+/datum/reagent/consumable/ethanol/abduction_fruit/on_mob_life(mob/living/carbon/M)
+	if(isabductor(M) || isxenoperson(M))
+		quality = RACE_DRINK
+	else
+		M.adjust_disgust(25)
+	return ..()
+
+/datum/reagent/consumable/ethanol/bug_zapper
+	name = "Bug Zapper"
+	color = "#F5882A" //(222, 250, 205)
+	description = "Metals and lemon juice. Hardly even a drink."
+	boozepwr = 5 //No booze really
+	taste_description = "copper and AC power"
+	glass_icon_state = "bug_zapper"
+	glass_name = "glass of Bug Zapper"
+	glass_desc = "An odd mix of copper, lemon juice and power meant for none-human consumption."
+
+/datum/reagent/consumable/ethanol/bug_zapper/on_mob_life(mob/living/carbon/M)
+	if(isinsect(M) || isflyperson(M) || ismoth(M))
+		quality = RACE_DRINK
+	else
+		M.adjust_disgust(25)
+	return ..()
+
+/datum/reagent/consumable/ethanol/mush_crush
+	name = "Mush Crush"
+	color = "#F5882A" //(222, 250, 205)
+	description = "Soil in a glass."
+	boozepwr = 5 //No booze really
+	taste_description = "dirt and iron"
+	glass_icon_state = "mush_crush"
+	glass_name = "glass of Mush Crush"
+	glass_desc = "Popular among people that want to grow their own food rather then drink the soil."
+
+/datum/reagent/consumable/ethanol/mush_crush/on_mob_life(mob/living/carbon/M)
+	if(ispodperson(M) || ismush(M))
+		quality = RACE_DRINK
+	else
+		M.adjust_disgust(25)
+	return ..()
+
+/datum/reagent/consumable/ethanol/darkbrew
+	name = "Darkbrew"
+	color = "#000000" //(0, 0, 0)
+	description = "Contained and acholic dark matter."
+	boozepwr = 5 //No booze really
+	taste_description = "Shadows and coffee trees"
+	glass_icon_state = "darkbrew"
+	glass_name = "glass of Darkbrew"
+	glass_desc = "A pitch black drink that's commonly confused with a type of coffee."
+
+/datum/reagent/consumable/ethanol/darkbrew/on_mob_life(mob/living/carbon/M)
+	if(isshadow(M))
+		quality = RACE_DRINK
+	else
+		M.adjust_disgust(25)
+	return ..()
+
+/datum/reagent/consumable/ethanol/hollow_bone
+	name = "Hollow Bone"
+	color = "#FCF7D4" //(252, 247, 212)
+	description = "Shockingly none-harmful mix of toxins and milk."
+	boozepwr = 15
+	taste_description = "Milk and salt"
+	glass_icon_state = "hollow_bone"
+	glass_name = "skull of Hollw Bone"
+	glass_desc = "Mixing of milk and bone hurting juice for enjoyment for rather skinny people."
+
+/datum/reagent/consumable/ethanol/hollow_bone/on_mob_life(mob/living/carbon/M)
+	if(isplasmaman(M) || isskeleton(M))
```

Comment:
can plasma men actually drink this? I think the helmet blocks.

Reply:
Take off hat and stuff they can. Also in 02-less area